tests/testthat/helper.R

```diff
@@ -140,9 +145,23 @@ dbg <- function(..., .display = utils::str) {
     cat(loc, "\n")
   } else {
     cat(sprintf("(from call: %s (srcfile missing))\n", trimws(
-      deparse1(sys.call(-2), width.cutoff = 60)
+      deparse1(sys.call(-2) %error% sys.call(-1), width.cutoff = 60)
     )))
   }
 
   invisible(out)
 }
+
+`%error%` <- function(x, y) tryCatch(x, error = function(e) y)
+
+# prevent new_class() from creating `S7::` prefixed S3 class names in tests.
+local({
```

Comment:
I wonder if it's better to be explicit this in `topNamespaceName()` by inlining then using `testthat::testing_package`?

Reply:
I don't understand, can you please explain more?

Reply:
Should the implementation of `topNamespaceName()` be this?

```R
topNamespaceName <- function(env = parent.frame()) {
  if (testthat::testing_package() == "S7") {
    return()
  }

  env <- topenv(env)
  if (isNamespace(env))
    getNamespaceName(env)
}
```

(but with `testthat::testing_package()` inlined into S7)